### ISSUE TITLE
Implement `ignore_index` logic when invoking `cross_entropy` calculation

### ISSUE THREAD

**Issue by nerdai** [nerdai/llms-from-scratch-rs#212]
To indicate that certain target tokens should not be used in the loss calculation, PyTorch uses `ignore_index` which is default to -100. We'll need to implement this logic in kind as well.
